src/Nethermind/Nethermind.BeaconNode.OApi/BeaconNodeOApiAdapter.cs

```diff
@@ -179,17 +179,17 @@ public async Task<ICollection<ValidatorDuty>> DutiesAsync(System.Collections.Gen
         {
             IEnumerable<BlsPublicKey> publicKeys = validator_pubkeys.Select(x => new BlsPublicKey(x));
             Epoch targetEpoch = epoch.HasValue ? new Epoch((ulong)epoch) : Epoch.None;
-            IList<BeaconNode.ValidatorDuty> duties = await _beaconNode.ValidatorDutiesAsync(publicKeys, targetEpoch);
-            List<ValidatorDuty> result = duties.Select(x =>
-                {
-                    ValidatorDuty validatorDuty = new ValidatorDuty();
-                    validatorDuty.Validator_pubkey = x.ValidatorPublicKey.Bytes;
-                    validatorDuty.Attestation_slot = (int)x.AttestationSlot;
-                    validatorDuty.Attestation_shard = (int)(ulong)x.AttestationShard;
-                    validatorDuty.Block_proposal_slot = x.BlockProposalSlot == Slot.None ? null : (int?)x.BlockProposalSlot;
-                    return validatorDuty;
-                })
-                .ToList();
+            var duties = _beaconNode.ValidatorDutiesAsync(publicKeys, targetEpoch);
+            List<ValidatorDuty> result = new List<ValidatorDuty>();
+            await foreach(var duty in duties)
```

Comment:
do you want to await foreach or await Task.WhenAll

Reply:
Probably mostly streamed for getting duties (e.g. a REST query, if over network, may need to batch due to GET path limitations). Parallel may be useful for get & sign block.